azurerm/internal/services/dns/dns_zone_data_source.go

```diff
@@ -109,26 +107,33 @@ func dataSourceArmDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 	return tags.FlattenAndSet(d, resp.Tags)
 }
 
-func findZone(client *dns.ZonesClient, rgClient *resources.GroupsClient, ctx context.Context, name string) (dns.Zone, string, error) {
-	groups, err := rgClient.List(ctx, "", nil)
+func findZone(client *dns.ZonesClient, ctx context.Context, name string) (dns.Zone, string, error) {
+	zonesIterator, err := client.ListComplete(ctx, nil)
 	if err != nil {
-		return dns.Zone{}, "", fmt.Errorf("Error listing Resource Groups: %+v", err)
+		return dns.Zone{}, "", fmt.Errorf("listing DNS Zones: %+v", err)
```

Comment:
that allows this to be nil - ensuring that if we return an error, trying to access the DNS Zone should panic - which'll be caught by tests (and forces us to check errors first):

```suggestion
		return nil, "", fmt.Errorf("listing DNS Zones: %+v", err)
```